### PR TITLE
In lighting, always normalize zero to (0, 0, 1)

### LINKED ISSUE #14167: [Android] Pro Yakyu Spirits 2010 (NPJH50234): Rendering errors with hardware transform off
### What happens?
Some Graphics are not appear correctly only Android. (EDIT: Turned out to be with hw transform off)
But, This issue will happen only Android.
(OpenGL & Vulkan both.)
Windows10 is not happen.

GEdump here(Got from Android).
 But, If you open the file on Windows  then it will seems correctly.
[NPJH50234_0004.zip](https://github.com/hrydgard/ppsspp/files/5993698/NPJH50234_0004.zip)

So, Please see attache screenshot.

**[Windows10 appear the Left & Right Team symbols correctly.]**
image

**[Android is missing the Left & Right Team symbols.]**
Screenshot_20210217-115050_PPSSPP

This BaseBall game has a other series.
- Pro Yakyu Spirits 2011
- Pro Yakyu Spirits 2012
- Pro Yakyu Spirits 2013
- Pro Yakyu Spirits 2014
**This problem** is not happen other series. Only 2010.
And, 2010 has a another graphic issue. #13391 

Plus,Other series(2011~2014) has another graphic glitch. #13980 :grin:

### What *should* happen?
Appear the graphic correctly.

### Please test newest version in https://buildbot.orphis.net/ppsspp/

I tested Windows & Android.

**[No Problem]**
PPSSPP Ver.：1.11.2-85
OS：Windows10（64-bit）Japanese
CPU：Ryzen7 3800X（8-core）
GPU：RADEON RADEON RX5700XT
Resolution：2560×1440
RAM：32GB

**[Have a Problem]**
PPSSPP Ver.：1.11.2-85
OS：Android 11.0
DeviceName：Samusung Galaxy S20 / SC-51A
CPU：Snapdragon865（Octa-core）
GPU：Adreno 650
Resolution：3200 x 1440 (Max)
RAM：12GB
ROM：128GB
ExternalStrage：microSDXC 512GB

### PR DESCRIPTION
**Pull request by unknownbrackets**
Fixes #14167.  We've seen several cases of this now, so I think it's better to just assume it's the case for all normalize usage.

-[Unknown]

### CODE REVIEW

**Comment by hrydgard** [comment]
I wonder if that SSE4 path is really worth it. But probably doesn't matter much.

**Comment by unknownbrackets** [comment]
Well, it was already there from #11425, I just figured might as well use it more places since it's there anyway.

-[Unknown]

**Comment by hrydgard** [comment]
Yeah, I know, just wondering out loud.

File: GPU/Common/VertexShaderGenerator.cpp [hunk 1]
```diff
@@ -1056,7 +1062,7 @@ bool GenerateVertexShader(const VShaderID &id, char *buffer, const ShaderLanguag
 						break;
 					case GE_PROJMAP_NORMALIZED_NORMAL:  // Use normalized transformed normal as source
 						if (hasNormal)
-							temp_tc = flipNormal ? "vec4(normalize(-normal), 1.0)" : "vec4(normalize(normal), 1.0)";
+							temp_tc = StringFromFormat("length(%snormal) == 0.0 ? vec4(0.0, 0.0, 1.0, 1.0) : vec4(normalize(%snormal), 1.0)", flipNormal ? "-" : "");
```
Comment (hrydgard) [inline GPU/Common/VertexShaderGenerator.cpp:1065]: @unknownbrackets I think the bug is here. Two %s but one parameter.

You don't really need the sign in the length check anyway...

Comment (unknownbrackets) [reply GPU/Common/VertexShaderGenerator.cpp:1065]: Oops, I don't think I intended to have it twice.  Bah.

-[Unknown]

Comment (hrydgard) [reply GPU/Common/VertexShaderGenerator.cpp:1065]: It's kinda ridiculous that this compiles.

Comment (unknownbrackets) [reply GPU/Common/VertexShaderGenerator.cpp:1065]: Makes me miss D where [this wouldn't](https://dlang.org/library/std/stdio/writef.html).  I'm sure rust has something similar.

-[Unknown]

Comment (hrydgard) [reply GPU/Common/VertexShaderGenerator.cpp:1065]: Indeed, this would break with Rust formatting macros.

**Pull request merged**